google-cloud-examples/src/main/java/com/google/cloud/examples/storage/snippets/StorageSnippets.java

```diff
@@ -1113,4 +1114,289 @@ public Bucket setDefaultKmsKey(String bucketName, String kmsKeyName) throws Stor
+
+    // The retetion period for objects in bucket
+    // 1 hour in seconds
+    // Long retentionPeriod = 3600L;
+
+    Bucket bucket =
+        storage.update(
+            BucketInfo.newBuilder(bucketName).setRetentionPeriod(retentionPeriod).build());
+
+    System.out.println(
+        "Retention period for " + bucketName + " is now " + bucket.getRetentionPeriod());
+    // [END storage_set_retention_policy]
+    return bucket;
+  }
+
+  /** Example of removing a retention policy on a bucket */
+  public Bucket removeRetentionPolicy(String bucketName) throws StorageException {
+    // [START storage_remove_retention_policy]
+    // Instantiate a Google Cloud Storage client
+    Storage storage = StorageOptions.getDefaultInstance().getService();
+
+    // The name of a bucket, e.g. "my-bucket"
+    // String bucketName = "my-bucket";
+
+    Bucket bucket = storage.get(bucketName, BucketGetOption.fields(BucketField.RETENTION_POLICY));
+    if (bucket.retentionPolicyIsLocked() != null && bucket.retentionPolicyIsLocked()) {
+      System.out.println("Unable to remove retention period as retention policy is locked.");
+      return bucket;
+    }
+
+    Bucket updated_bucket = bucket.toBuilder().setRetentionPeriod(null).build().update();
+
+    System.out.println("Retention period for " + bucketName + " has been removed");
+    // [END storage_remove_retention_policy]
+    return updated_bucket;
+  }
+
+  /** Example of removing a retention policy on a bucket */
+  public Bucket getRetentionPolicy(String bucketName) throws StorageException {
+    // [START storage_get_retention_policy]
+    // Instantiate a Google Cloud Storage client
+    Storage storage = StorageOptions.getDefaultInstance().getService();
+
+    // The name of a bucket, e.g. "my-bucket"
+    // String bucketName = "my-bucket";
+
+    Bucket bucket = storage.get(bucketName, BucketGetOption.fields(BucketField.RETENTION_POLICY));
+
+    System.out.println("Retention Policy for " + bucketName);
+
+    System.out.println("Retention Period: " + bucket.getRetentionPeriod());
+    if (bucket.retentionPolicyIsLocked() != null && bucket.retentionPolicyIsLocked()) {
+      System.out.println("Retention Policy is locked");
+    }
+    if (bucket.getRetentionEffectiveTime() != null) {
+      System.out.println("Effective Time: " + new Date(bucket.getRetentionEffectiveTime()));
+    }
+    // [END storage_get_retention_policy]
+    return bucket;
+  }
+
+  /** Example of how to lock a bucket retention policy */
+  public Bucket lockRetentionPolicy(String bucketName) throws StorageException {
+    // [START storage_lock_retention_policy]
+    // Instantiate a Google Cloud Storage client
+    Storage storage = StorageOptions.getDefaultInstance().getService();
+
+    // The name of a bucket, e.g. "my-bucket"
+    // String bucketName = "my-bucket";
+
+    Bucket bucket =
+        storage.get(bucketName, Storage.BucketGetOption.fields(BucketField.METAGENERATION));
+    Bucket updated_bucket =
```

Comment:
Do you need to call `bucket.update` here?

Reply:
No `lockRetentionPolicy()` is a new RPC.